src/actions/generalActions.js

```diff
@@ -0,0 +1,57 @@
+const reactToGeneralAction = (model, toolHandler) => {
+	return (action) => {
+		switch (action.type) {
+			case 'MODE_UPDATE':
+				if (model.activityInProgress === true) {
+          toolHandler.cleanTool();
+          model.activityInProgress = false;
+        }
+
+        if (model.mode !== action.mode) {
+          model.mode = action.mode;
+          toolHandler.updateTool();
+        }
+				break;
+          
+      // UNCALLED
+			case 'ACTIVITY_UPDATE':
+				model.activityInProgress = action.inProgress;
+				break;
+
+      case 'ANNOTATIONS_RESET':
+        model.activityInProgress = false;
+        model.annotations = action.annotations || [];
+    
+        break;
+        
+      case 'SELECTION_RESET':
+        model.activityInProgress = false;
+        model.selection = null;
+    
+        break;
+    
+      case 'ZOOM_UPDATE':
+        if (model.activityInProgress) {
+          toolHandler.handleAction('ZOOM_UPDATE');
+        }
+    
+        model.zoom = action.zoom;
```

Comment:
what is the difference between calling toolHandler.handleAction('zoomupdate') and calling model.zoomupdate() ?

Reply:
I'll eventually remove `model.zoomUpdate`, it exists purely for the text annotation feature that we don't use. That function call scales any text annotations' font size. We can ignore that right now, I'll remove it before the code is ready. 

I'm passing the zoomupdate to the toolhandler as some tools react differently upon zoom. For example - 
- For all tools, activity stream is broken (activityInProgress set to false), and clicks are reset on zoom. 
- For polygon or freedraw, we break out of annotation creation and remove/delete the creation in progress if the user accidentally zooms while creating the annotation.

Reply:
Interesting. We had this issue with the accept/reject on the review ROI and then went to a icon based approach due to scaling problems with text. Good to know this can be solved if we need to bring text back in the future

Reply:
Noted. Have kept the original text zoom function  intact, we can repurpose if the need arises